### ISSUE TITLE
Self Signed SSL Cert Errors

### ISSUE THREAD

**Issue by DrSeussFreak** [causefx/Organizr#1177]
###### Operating System: Ubuntu 18.04 LTS
<hr>

##### Problem Description:

I have self signed SSL certs, and there is no way to tell Organizr to ignore certificate warnings.  I want to continue using my certs, and have Organizr function.

<hr>
##### Reproduction Steps:

Hit test on plex with self signed SSL Cert installed

<hr>
#### Errors on screen?  If so paste here:

```
<img width="713" alt="Screen Shot 2019-05-18 at 9 10 06 PM" src="https://user-images.githubusercontent.com/175500/57976906-55f80280-79b1-11e9-8f52-88c59da95834.png">
```

**Comment by DrSeussFreak** [comment]
TO be clear, this is when testing my connection to plex

**Comment by Roxedus** [comment]
Replace the cert in `<OrgDir>\api\functions\cert` with your own

**Comment by Marc-Vieg** [comment]
hello, I'm having kind of the same issue. Organizr works with two hosts : 
hots A : Emby, jellyfin, ombi
host B : radarr, sonarr, lidarr
I replaced the certificate in `<OrgDir>\api\functions\cert` with cert from host B, Emby, Jellyfin, Ombi are now working.
But I tried to append the cert from host A  in this file but it trows error when accessing  radarr/sonarr/lidarr
can you tell me how to give Organizr multiple self signed certs ? 
Cheers

**Comment by jackyaz** [comment]
Sorry, what name(s) should i use for the cert in the mentioned directory?

```
jack@DockerBox:/apps/organizr/www/Dashboard/api/functions/cert$ ls
index.html
```

Do I use the full chain, the cert/key for Organizr, or the CA?

**Comment by causefx** [comment]
I think I will just add cert ignore for the meantime until I can adopt cert inputs for each items.  The tricky one will be sonarr as I use an existing class - i can just edit that I suppose.

**Comment by darkcloud784** [comment]
I also have this issue within the qbittorrent module. I run everything in kubernetes and don't like to use all my node ports so I try to conserve through using clusterip and resolve it to port on the ingress using an internal domain.

**Comment by maichai** [comment]
This issue can practically happen anywhere (another example where I have this: SabNZBD - here the problem is actually with curl, where does curl read the root CA certs?)! The ignore button should be there for all homepage connections.

Actually I added my own root CA's certificate to ...api/functions/cert/cacert.pem, but it that did not solve the problem with self signed certificates (remember: every! root CA certificate is selfsigned!)

There is probably also a bug in the evaluation of the cacert.pem

**Comment by causefx** [comment]
we will keep this open as i still have plans for this.

**Comment by causefx** [comment]
I still have plans for this.

**Comment by tdylewski83** [comment]
I was able to get it to work with some services by:
1. creating a custom file at **api/pages/custom/setSelfSignedCert.php**

2. I put this into the file
`<?php
Requests::set_certificate_path('/config/www/organizr/cert/yourCert.pem');`

3. I made sure the file **/config/www/organizr/cert/yourCert.pem** exists :)
and it has all the important certs inside (http://curl.haxx.se/ca/cacert.pem + my root CA)

Now my self-signed self-hosted services can be accessed without issues.

**Comment by maichai** [comment]
+1, there should be either options to ignore the self-signed cert error, or ways to provide an own root ca cert to be used i all modules.

I am having this problem with SabNZB and Deluge.

**Comment by tdylewski83** [comment]
@maichai if I'm not mistaken deluge connection needs the non-existent cert file at `<OrgDir>/cert/cacert.pem` OR `<OrgDir>\api\functions\cert\cacert.pem`
try creating the path and the file with your own cert (just remember to merge it with http://curl.haxx.se/ca/cacert.pem)

**Comment by causefx** [comment]
Next week - I will be overhauling all the homepage settings items to include options to use self-signed certs and ignore cert validation.

**Comment by causefx** [comment]
I went ahead and started working on this.  I have planned on how I want to do it and have converted over home homepage items.  I have completed about 5 items so far.  

Using Organizr certs:
![image](https://user-images.githubusercontent.com/16184466/130540644-6eb0a408-d8f9-4a54-ab9d-0c05ae6ada02.png)

Using custom cert:
![image](https://user-images.githubusercontent.com/16184466/130540678-07952854-c1fe-476b-bce4-cf702cca0365.png)
